src/github/client/github-client.ts

```diff
@@ -28,7 +28,7 @@ import { getLogger } from "config/logger";
 
 export interface GitHubClientApiKeyConfig {
 	headerName: string;
-	apiKeyGenerator: () => Promise<string>;
+	apiKeyGenerator: () => Promise<string | undefined>;
```

Comment:
Not quite sure I understand this... API key config consists of 2 items: header and value. Either both should be absent (then the object that holds a reference to the config should contain null/undefined), or present. In both cases, I think apiKeyGenerator should always return a string